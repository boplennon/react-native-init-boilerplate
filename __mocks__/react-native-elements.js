
module.exports = ({
  FormValidationMessage: 'View',
  SocialIcon: 'Button',
  FormLabel: 'Text',
  Icon: 'View'
});

