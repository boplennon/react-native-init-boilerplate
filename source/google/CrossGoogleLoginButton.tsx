import React from 'react';
import { SocialIcon } from 'react-native-elements';
import { GoogleSignin } from 'react-native-google-signin';
import { Platform, StyleProp, ViewStyle } from 'react-native';

const configParam =
  Platform.OS === 'ios'
    ? {
        iosClientId: 'TODO: Settings',
      }
    : {};

    export interface ICrossGoogleLoginButtonProps {
      color?: string;
      style?: StyleProp<ViewStyle>;
    }

/**
 * Google login button using React Native Elements button and `react-native-google-signin`
 */
export class CrossGoogleLoginButton extends React.Component<ICrossGoogleLoginButtonProps> {
  async onPress() {
    await GoogleSignin.configure({
      ...configParam,
      webClientId: 'TODO: Settings',
      offlineAccess: false,
    });

    // Google native signin

    const result = await GoogleSignin.signIn();

    console.log('** Google Signin **');
    console.log(result);
  }

  render() {
    // Social icon's click works badly in debug so we return a normal button
    // if (__DEV__)
    //     return <Button title='Google debug' onPress={() => onPress()} />;
    return (
      <SocialIcon
      {...this.props}
        title='Log in with Google'
        button
        onPress={this.onPress.bind(this)}
        type='google-plus-official'
      />
    );
  }
}

export default CrossGoogleLoginButton;
