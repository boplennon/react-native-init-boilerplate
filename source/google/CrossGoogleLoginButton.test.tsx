/// <reference types="jest"/>
import React from 'react';
import { shallow } from 'enzyme';
import { SocialIcon } from 'react-native-elements';
import CrossGoogleLoginButton from './CrossGoogleLoginButton';

jest.unmock('react-native');
jest.unmock('react-native-elements');

function setup() {
  // @ts-ignore - Enzyme TypeScript issues
  const enzymeWrapper = shallow(
    <CrossGoogleLoginButton />
  );

  return {
    enzymeWrapper,
  };
}

describe('components', () => {
  describe('GoogleLogin', () => {
    const { enzymeWrapper } = setup();
    const loginBase = enzymeWrapper;

    it('should match snapshot', () => {
      expect(loginBase).toMatchSnapshot();
    });

    it('type should be google-plus-official', () => {
      const buttonGoogle = loginBase
        .find(SocialIcon)
        .first()
        .props();
      expect(buttonGoogle.type).toEqual('google-plus-official');
    });

    it('should be able to press', () => {
      const buttonGoogle = loginBase.find(SocialIcon).first();
      buttonGoogle.simulate('onPress');
    });
  });
});
