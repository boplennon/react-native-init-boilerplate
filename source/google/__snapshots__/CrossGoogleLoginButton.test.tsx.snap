// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components GoogleLogin should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CrossGoogleLoginButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "button": true,
      "iconColor": "white",
      "iconSize": 24,
      "onPress": [Function],
      "raised": true,
      "title": "Log in with Google",
      "type": "google-plus-official",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "button": true,
        "iconColor": "white",
        "iconSize": 24,
        "onPress": [Function],
        "raised": true,
        "title": "Log in with Google",
        "type": "google-plus-official",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
